cfbs/commands.py

```diff
@@ -1377,6 +1381,74 @@ def cfbs_convert_git_commit(
                 log.warning("Git commit failed, continuing without committing...")
         if response == "2":
             print("Keeping file as is, nothing to do.")
+        if response == "3":
+            print("Converting './%s' into a patch file..." % modified_file)
+            patches_dir = "custom-masterfiles-patches"
+            patches_module = "./" + patches_dir + "/"
+
+            actual_path_modified_file = os.path.join(dir_name, modified_file)
+            actual_path_original_file = mpf_filepath
+
+            file_diff_data = file_diff_text(
+                actual_path_original_file,
+                actual_path_modified_file,
+                modified_file,
+                modified_file,
+            )
+
+            patch_filename = modified_file.replace("/", "_") + ".patch"
+            patch_path = os.path.join(patches_dir, patch_filename)
+            # append a number if there are multiple files with the same filename
+            i = 1
+            while os.path.exists(patch_path):
+                patch_filename = (
+                    modified_file.replace("/", "_") + "-" + str(i) + ".patch"
+                )
+                patch_path = os.path.join(patches_dir, patch_filename)
+                i += 1
+
+            # saving the .patch file should be done before creating the patches module
+            try:
+                save_file(patch_path, file_diff_data)
+            except:
+                log.warning(
+                    "Saving the patch file failed - keeping the modified file as is instead and continuing..."
+                )
+
+            # make the patches local module on first use
+            if not patches_module_present:
+                print("Adding patches local module...")
+                mkdir(patches_dir)
+                try:
+                    r = add_command(
+                        [patches_module],
+                        added_by="cfbs convert",
+                        explicit_build_steps=["patch " + patch_filename],
+                    )
+                    # `explicit_build_steps=[]` would fail validation
+                except Exception as e:
+                    log.warning(
+                        "Adding the patches local module failed (%s), continuing..."
+                        % str(e)
+                    )
+                if r != 0:
+                    log.warning("Adding the patches local module failed, continuing...")
```

Comment:
Is it safe to continue? At the very least, `patches_module_present` should probably not be set to `True` on the next instruction?

Reply:
Another thing is that this warning might not be seen by the user if there is a lot of other output. And they could end up with policy that does not work.